### ISSUE TITLE
[ASImageNode] strange behaviour with UIViewContentModeScaleAspectFill 

### ISSUE THREAD

**Issue by pashaisthebest** [facebook/AsyncDisplayKit#1612]
initialisation code 

```
_avatarNode = [ASImageNode new];
_avatarNode.layerBacked = YES;
_avatarNode.contentMode = UIViewContentModeScaleAspectFill;
[self addSubnode:_avatarNode];
```

layout

```
- (void)layoutSubviews {
    [super layoutSubviews];
    _avatarNode.frame = self.bounds;
}
```

result - https://youtu.be/jDM7EDGhvZw

Tried to set layerBacked = NO, did not help.
Will look into the problem too, if I can find what is causing it, I will fix it and do a pull request. If there is any additional info you need - I can provide it, probably even create some sample project if needed.

**Comment by appleguy** [comment]
Thanks for the video!  This looks more like an issue with your layout code than the image drawing, but that's not certain - not quite enough detail.

You should probably handle this interaction by manually setting the frame so that it scales in the right way, as it would be extremely inefficient to trigger redrawing on every frame.  So needsDisplayOnBoundsChange should be NO, and you should be setting frame values that extend off the left and right edges of the screen when it is in the "over-bounce" state.

**Comment by pashaisthebest** [comment]
Just to make sure that it is an issue on my side, I created this sample project for you to check out.
https://github.com/pashaisthebest/ASImageNodeTest

Reproduction steps are just like the steps that you can see in the video, that I posted earlier.
The frame of the node is shown in green. I also noticed, that UIViewContentModeScaleAspectFill is just not working the way I expect it to for this image for example:

https://pp.vk.me/c627530/v627530193/3ddfb/tgKrnJGP51w.jpg 

When I pull down the table I expect to see something like this:
![simulator screen shot 03 may 2016 09 46 23 2](https://cloud.githubusercontent.com/assets/2697487/14976432/bcb4a18c-111c-11e6-973c-628854bafb54.png)

But instead I get:
![simulator screen shot 03 may 2016 09 46 23](https://cloud.githubusercontent.com/assets/2697487/14976439/c4355e74-111c-11e6-867d-a315e4e8196e.png)

**Comment by appleguy** [comment]
Thanks for creating a reproduction case, that is tremendously helpful!  @garrettmoon, I know you have several urgent tasks on your plate this week, but could you investigate this next week or with a 1hr initial investigation this week?

**Comment by garrettmoon** [comment]
I'll try and take a look tomorrow. If I remember correctly, I think this may not be a regression, but not currently supported. I'll confirm tomorrow though.

**Comment by garrettmoon** [comment]
Two things. First of all, I was incorrect, ASImageNode *does* support this contentMode (but not some others).

What's going on is that the image is being drawn at the original size and then when bounds change, it's being scaled. So it's actually behaving as expected. But that's obviously not what you're looking for.

TL;DR; turn off cropping:

```
_avatarNode.cropEnabled = NO;
```

**Comment by pashaisthebest** [comment]
Thanks for the time! I did not know that this cropping was turned on by default.
Now it is working just like I want it to.
